src/main/java/com/google/devtools/build/lib/bazel/bzlmod/BazelLockFileValue.java

```diff
@@ -26,9 +26,17 @@
 import java.util.Optional;
 
 /**
- * The result of reading the lockfile. Contains the lockfile version, module hash, definitions of
- * module repositories, post-resolution dependency graph and module extensions data (ID, hash,
- * definition, usages)
+ * The result of reading a lockfile. Contains the lockfile version as well as registry and module
+ * extensions data (ID, usages hash, generated repos, ...).
+ *
+ * <p>Bazel maintains two separate lockfiles:
+ *
+ * <ul>
+ *   <li>the (regular) lockfile stored as MODULE.bazel.lock under the workspace directory;
+ *   <li>the persistent lockfile stored as MODULE.bazel.lock under the output base.
```

Comment:
I find it a bit weird to call this one the "persistent" lockfile. If anything, it's usually _less_ "persistent" than the lockfile in the workspace, since it's not checked into VCS, and easily wiped by `bazel clean`.

This lockfile is generated only for reproducible extensions, right? Maybe "reproducible lockfile" (ok that's bad). Or, part of the whole point is that it's not user-visible. Maybe "hidden lockfile"?

Reply:
It's only wiped out by a `bazel clean --expunge`, just like the persistent action cache (which I named this after). But I like "hidden" better.